docs/how-tos/deploy-ml.md

```diff
@@ -24,28 +28,92 @@ You can use Viam's built-in tools to train a machine learning (ML) model on your
 ![Diagram of the camera component to data management service to ML model service to vision service pipeline.](/how-tos/ml-vision-diagram.png)
 
 For example, you can train a model to recognize your dog and detect whether they are sitting or standing.
-Then, you can configure your machine to [capture images](/how-tos/image-data/) only when your dog is in the camera frame so you don't capture hundreds of photos of an empty room.
-You can then get even more image data of your dog and improve your ML model by training it on the larger dataset.
-
-You can do all of this using the [Viam app](https://app.viam.com) user interface.
-You will not need to write any code.
+You could then use that knowledge to [give your dog treats](https://www.viam.com/post/smart-pet-feeder) or [capture images](/tutorials/configure/pet-photographer/) only when your dog is in the camera frame so you don't capture hundreds of photos of an empty room.
 
 {{< alert title="In this page" color="tip" >}}
 
 1. [Create a dataset and label data](#create-a-dataset-and-label-data)
-2. [Train a machine learning (ML) model](#train-a-machine-learning-ml-model)
-3. [Deploy a machine learning model](#deploy-an-ml-model)
+1. [Train a machine learning (ML) model](#train-a-machine-learning-ml-model)
+1. [Test your ML model](#test-your-ml-model)
+1. [Deploy your ML model](#deploy-an-ml-model)
+
+{{< /alert >}}
+
+## Prerequisites
+
+{{% expand "A running machine connected to the Viam app. Click to see instructions." %}}
+
+{{% snippet "setup.md" %}}
+
+{{% /expand%}}
+
+{{% expand "A configured camera. Click to see instructions." %}}
+
+Navigate to the **CONFIGURE** tab of your machine's page in [the Viam app](https://app.viam.com).
+Click the **+** icon next to your machine part in the left-hand menu and select **Component**.
+Then [find and add a camera model](/components/camera/) that supports your camera.
+
+If you are not sure what to use, start with a [webcam](/components/camera/webcam/) which supports most USB cameras and inbuilt laptop webcams.
+
+{{% /expand%}}
+
+{{% expand "No computer or webcam?" %}}
+No problem.
+You don't need to buy or own any hardware to complete this guide.
 
+Use [Try Viam](https://app.viam.com/try) to borrow a rover free of cost online.
+The rover already has `viam-server` installed and is configured with some components to test with, including a webcam.
+
+Once you have borrowed a rover, go to the **CONFIGURE** tab of the machine, find the cameras and click on the **Test** panel at the bottom of each camera's configuration panel to test the camera stream.
+You should have a front-facing camera and an overhead view of your rover.
+Now you know what the rover can _perceive_.
+
+If your rover is facing a wall, find the base configuration panel and click on its **Test** panel.
+Use the controls to drive your rover to a different location.
```

Comment:
Need to tell them to turn on picture-in-picture (either from config or control).

What's wrong with it facing a wall? It's technically always going to be facing a wall :D I think this is confusing because we don't tell them what the goal is.

```suggestion
To change what the front-facing camera is pointed at, find the **cam** camera panel on the **CONTROL** tab and click **Toggle picture-in-picture** so you can continue to view the camera stream.
Then, find the **viam_base** panel and drive the rover around.
```